documentation/milestones/individual_milestones/api_planning.md

```diff
@@ -0,0 +1,224 @@
+
+### API Framework
+
+Evaluate and select the web framework or library we'll use to implement the API in the chosen language.
+
+Some potential options include:
+
+- Flask or FastAPI for Python
+- Express.js or Fastify for Node.js
+- Gin or Fiber for Go
+- Others recommended by the dev team
+
+Some factors to consider are:
+
+- Is this framework popular with a broad community of users?
+- Is this framework actively maintained, with a strong governance structure?
+- How does this framework support important features like concurrency, serialization, middleware?
+- Is there an established set of plugins or libraries that extend the framework's key features? (e.g. AuthN/AuthZ, API documentation, etc.)
+
+### Type of API
+
+Evaluate and select the API paradigm our endpoints will adhere to.
+
+Some potential options include:
+
+- REST
+- SOAP
+- GraphQL
+- RPC
+- Others recommended by the dev team
+
+Some factors to consider are:
+
+- What functionality does our API need to support (e.g. recurring and predictable data requests, highly variable data queries, execution of internal functions, etc.)? Are certain API types better at supporting this functionality than others?
+- Which API paradigms are most common? Are certain types of APIs easier for developers to learn and interact with than others?
+- How often do we expect either the implementation or the interface of a given endpoint to change? Is it easier to make these changes with certain types of APIs than others?
+- What are our needs around API performance? Do certain types of APIs handle performance optimization better than others?
+
+### API Deployment Service
+
+Evaluate and select an option used to deploy and host the API.
+
+Some potential options include:
+
+- Dedicated EC2 instance or ECS cluster
+- AWS Fargate
+- AWS Lambda + API Gateway
+- Others recommended by dev team
+
+Some factors to consider are:
+
+- Is this deployment option FedRAMP compliant?
+- How commonly used is this deployment option? Is it easy for new developers to learn?
+- How expensive is it to deploy an API with this option, both in terms of direct operating costs and in terms of team resources?
+- How easily does this option scale?
+
+### Definition of done
+<!-- Required -->
+
+- [ ] The tools and services chosen have been authorized under the Grants.gov ATO (Authority to Operate)
+- [ ] An ADR has been recorded and published to `main` for each of the following choices:
+  - [ ] API Language
+  - [ ] API Framework
+  - [ ] API Type
+  - [ ] API Deployment Service
+- [ ] Architecture diagram
+  - [ ] An initial architecture diagram for the backend has been published to the `main` branch of the repository
+  - [ ] The diagram is stored in a format that supports tracking individual changes with version control
+
+### Proposed metrics for measuring goals/value/definition of done
```

Comment:
I agree. There may be a point of diminishing returns on what we understand vs. how long we spend to collect the metrics. We can maybe note it on the ticket level, cap the time spent researching or give justification on why we may not have gotten a certain metric.

Reply:
I agree as well. Maybe this could renamed as "Potential metrics" or "Optional metrics" to indicate that they are not required,

Reply:
Yup. For services going live into production, the metrics are very important to me, but for planning milestones, not so much.

Reply:
@lucasmbrown-usds @acouch @sumiat These are excellent suggestions, I wound up splitting this section into two parts: 

- One-time Metrics
- Ongoing Metrics

The two metrics I preserved as ongoing metrics, I feel are pretty important for us to track about all of the top-level libraries we're selecting as dependencies, and they could be pulled pretty easily from the GitHub API or from the package manager APIs as well (e.g. PyPI and NPM)

I know we have so much in flight but perhaps it would be worth prioritizing the measurement framework milestone so that rather than picking metrics per deliverable in each milestone (which is what I have been doing), we instead have top-level metric categories that these milestone metrics rollup into. 

For example:

- **Dependency Health** - Metrics about the maintenance and security of project dependencies
- **Development Health** - Metrics about deployment velocity, burndown, etc.
- **Community Health** - Metrics about open source engagement, meetings, etc.